main.cpp

```diff
@@ -19,13 +19,23 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+	auto l_brace = cstmtptr->getLParenLoc();
+	auto r_brace = cstmtptr->getRParenLoc();
+	auto end_of_var = Lexer::getLocForEndOfToken(cstmtptr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(), 0, *Result.SourceManager, LangOptions());
+	auto start_of_var = cstmtptr->getSubExprAsWritten()->getBeginLoc();
+	_rewriter.RemoveText(l_brace, 1);
+	_rewriter.ReplaceText(CharSourceRange::getCharRange(r_brace, start_of_var),">");

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()`.

Reply:
getCharRange() returns range from the starting to ending character by the specified locations.
getTokenRange() returns range specifying the start of the range and the start of the last token.
getTokenRange() will return all tokens that are in the specified range when getCharRange() returns only a string contained in that range.

Reply:
Examples required.

Reply:
To see the difference more clearly we can run something like:

  auto charRange = Lexer::getAsCharRange(
      CharSourceRange::getCharRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  auto tokenRange = Lexer::getAsCharRange(
      CharSourceRange::getTokenRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  llvm::outs() << "\n char range start: " << charRange.getBegin().printToString(SM)
                     << ",\n char range end: " << charRange.getEnd().printToString(SM) << "\n";

  llvm::outs() << "\n token range start: " << tokenRange.getBegin().printToString(SM)
                     << ",\n token range end: " << tokenRange.getEnd().printToString(SM) << "\n";
on this code:

 int main() {
     int i = (int)1.234;
     return 0;
 }
The result will be:

   char range start: /UNN-C-Style-Checker/test/test2.cpp:2:14,
   char range end: /UNN-C-Style-Checker/test/test2.cpp:2:15
  
   token range start: /UNN-C-Style-Checker/test/test2.cpp:2:14,
   token range end: /UNN-C-Style-Checker/test/test2.cpp:2:17